l10n_ch_pain_base/models/account_move_line.py

```diff
@@ -0,0 +1,20 @@
+# copyright 2016 Akretion (Alexis de Lattre <alexis.delattre@akretion.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models
+
+
+class AccountMoveLine(models.Model):
+    _inherit = "account.move.line"
+
+    def _prepare_payment_line_vals(self, payment_order):
+        vals = super()._prepare_payment_line_vals(payment_order)
+        if self.move_id.partner_bank_id and self.move_id._has_isr_ref():
+            if self.move_id.partner_bank_id._is_qr_iban():
+                vals["communication_type"] = "qrr"
```

Comment:
In QRR case, the QR reference should be communicated to creditor, see https://github.com/SimoneVagile/l10n-switzerland/pull/1

Reply:
Merged, thanks